README.md

```diff
@@ -1,81 +1,192 @@
+
+# This ensures HVM is correctly installed and accessible.
+hvm --version
 ```
+2. Install Bend by running:
+```sh
+# This command will install Bend
+cargo install bend-lang
 
-Finally, write some Bend file, and run it with one of these commands:
+# This ensures Bend is correctly installed and accessible.
+bend --version
+```
 
+### Getting Started
+#### Running Bend Programs
 ```sh
 bend run    <file.bend> # uses the Rust interpreter (sequential)
 bend run-c  <file.bend> # uses the C interpreter (parallel)
 bend run-cu <file.bend> # uses the CUDA interpreter (massively parallel)
+
+# Notes
+# You can also compile Bend to standalone C/CUDA files using gen-c and gen-cu for maximum performance.
+# The code generator is still in its early stages and not as mature as compilers like GCC and GHC.
+# You can use the -s flag to have more information on
+  # Reductions
+  # Time the code took to run
+  # Interaction per second (In millions)
 ```
 
-You can also compile `Bend` to standalone C/CUDA files with `gen-c` and
-`gen-cu`, for maximum performance. But keep in mind our code gen is still in its
-infancy, and is nowhere as mature as SOTA compilers like GCC and GHC.
+#### Testing Bend Programs
+The example below sums all the numbers in the range from `start` to `target`. It can be written in two different methods: one that is inherently sequential (and thus cannot be parallelized), and another that is easily parallelizable. (We will be using the `-s`flag in most examples, for the sake of visibility)
+
+#### Sequential version:
+First, create a file named `ssum.bend`
+```sh
+# Write this command on your terminal
+touch ssum.bend
+```
+Then with your text editor, open the file `ssum.bend`, copy the code below and paste in the file.
+
+```py
+# Defines the function Sum with two parameters: start and target
+def Sum(start, target):
+    # If the value of start is the same as target, returns start
+  if start == target:
+    return start
+    # If start is not equal to target, recursively call Sum with start incremented by 1, and add the result to start
+  else:
+    return start + Sum(start + 1, target)  
+
+def main():
+# This translates to (1 + (2 + (3 + (...... + (79999999 + 80000000)))))
+  return Sum(1, 80000000)
+```
+
+##### Running the file
+You can run it using Rust interpreter (Sequential)
+```sh
+bend run ssum.bend -s
+```
+
+Or you can run it using C interpreter (Sequential)
+```sh
+bend run-c ssum.bend -s
+```
+
+If you have a NVIDIA GPU, you can also run in CUDA (Sequential)
+```sh
+bend run-cu ssum.bend -s
+```
+
+In this version, the next value to be calculated depends on the previous sum, meaning that it cannot proceed until the current computation is complete. Now, let's look at the easily parallelizable version.
+
 
-## Parallel Programming in Bend
+#### Parallelizable version:
+First close the old file and then proceed to your terminal to create `psum.bend`
+```sh
+# Write this command on your terminal
+touch psum.bend
```

Comment:
use a decent name for the files (parallel_sum, sequential_sum, etc)

Reply:
Okay, will edit